### DISCUSSION TITLE
File Ascending Sort Not Working Properly with mix of numbers and letters

### DISCUSSION THREAD

**Discussion by yurijserrano** [RafaelGB/obsidian-db-folder#630] [Ideas]
### Contact Details

yurijserrano@gmail.com

### What happened?

Good Evening @RafaelGB 

I hope you are well

When I select the option to sort the results by the filename in ascending order, the results do not appear in the correct order, as can be seen in the video present in the link below:

**[BUG - VIDEO](https://share.getcloudapp.com/mXuboqvz)**

If you need anything else, just let me know.

I am waiting to hear from you.

### What platform were you using?

Desktop

### Version of the plugin

2.9.2

### Are you using the latest version of the plugin in your Obsidian vault?

- [X] I have verified that I am on the latest version

### Are you check if there is a similar issue?

- [X] I have verified that there are not similar issues

**Comment by RafaelGB** [comment]
the problem is that an alphabetical order is used.

"file9">"file109"
"file1"<"file109"

You can use a padding of Zeros to fix it

"file009"<"file109"

Internally the plugin can not know how to order correctly with a mix of strings and numbers

**Comment by RafaelGB** [comment]
the way to fix this without padding would be to check when a string ends with numbers, but that can slow down performance a lot

**Comment by yurijserrano** [comment]
Good Morning @RafaelGB 

Regarding this excerpt:

> You can use a padding of Zeros to fix it

The problem with this approach is that I will always have to change the notes, so it is not the best scenario.

Regarding this excerpt:

> the way to fix this without padding would be to check when a string ends with numbers, but that can slow down performance a lot

I had an idea to solve this problem and keep a good performance, after some tests if it works I will send a pull request.

But meanwhile I managed to get the correct order by sorting first by date and then by filename.

Thank you so much for your attention to this matter @RafaelGB 

Have a wonderful day.
